db/model/subject.js

```diff
@@ -431,6 +425,11 @@ module.exports = function subject(seq, dataTypes) {
         } else if (inst.previous('isPublished')) {
           // Treat unpublishing a subject as a "delete" event.
           common.publishChange(inst, eventName.del);
+
+          if (featureToggles.isFeatureEnabled(sampleStoreFeature)) {
+            subjectUtils.removeFromRedis(inst.absolutePath);
+          }
```

Comment:
Do we need this here too? I think the call to removeFromRedis at 354 will take care of this case.